registry/config.go

```diff
@@ -379,10 +375,44 @@ func ParseIndexInfo(reposName string) (*IndexInfo, error) {
 
 // NormalizeLocalName transforms a repository name into a normalize LocalName
 // Passes through the name without transformation on error (image id, etc)
+// It does not use the repository info because we don't want to load
+// the repository index and do request over the network.
 func NormalizeLocalName(name string) string {
-	repoInfo, err := ParseRepositoryInfo(name)
+	indexName, remoteName := splitReposName(name)
```

Comment:
I feel like I left it out for some reason. But I don't think it can harm. I've changed the code a little bit to reuse a validate function that we already implemented with the checks we need.